### PR TITLE
fix: don't download collectlogs.ps1 if it exists

### LINKED ISSUE #2986: collect-windows-logs.ps1 downloads script from GitHub
**Describe the bug**
The [collect-windows-logs.ps1](https://github.com/Azure/aks-engine/blob/master/scripts/collect-windows-logs.ps1) script is used to gather logs and debugging information off Windows nodes. Part of this script downloads another debugging script directly from GitHub, to collect networking logs [collectlogs.ps1](https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/debug/collectlogs.ps1)

**Expected behavior**
I would expect to not have a script download untrusted content directly off GitHub. I suggest having a signed script being present on the VHD at provision time. For customers running in locked down environments, it's not certain that there is access to the script from the Windows host.

**Additional context**
Currently this dependency is not documented for [AKS in the egress lockdown document](
https://docs.microsoft.com/en-us/azure/aks/limit-egress-traffic#required-by-windows-server-based-nodes-in-public-preview-enabled
), and we would like to to add this as an external dependency.

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

Updating behavior for collect-windows-logs.ps1 to check if collectlogs.ps1 is present on disk before trying to download it.

**Issue Fixed**:

Fixes #2986 

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: scripts/collect-windows-logs.ps1 [hunk 1]
```diff
@@ -36,12 +38,17 @@ Get-CimInstance win32_pagefileusage | Format-List * | Out-File -Append "$ENV:TEM
 Get-CimInstance win32_computersystem | Format-List AutomaticManagedPagefile | Out-File -Append "$ENV:TEMP\\$($timeStamp)_pagefile.txt"
 $paths += "$ENV:TEMP\\$($timeStamp)_pagefile.txt"
 mkdir 'c:\k\debug' -ErrorAction Ignore | Out-Null
-Invoke-WebRequest -UseBasicParsing https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/debug/collectlogs.ps1 -OutFile 'c:\k\debug\collectlogs.ps1'
+
+Write-Host "Collecting networking related logs"
+if (-not (Test-Path 'c:\k\debug\collectlogs.ps1')) {
+  Invoke-WebRequest -UseBasicParsing https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/debug/collectlogs.ps1 -OutFile 'c:\k\debug\collectlogs.ps1'
```
Comment (AbelHu) [inline scripts/collect-windows-logs.ps1:44]: Could we add some logs to measure how long it takes to download the file?

Comment (marosset) [reply scripts/collect-windows-logs.ps1:44]: This script is isn't called during CSE execution / provisioning.
Do you just want the time written to stdout/write-host?

Comment (AbelHu) [reply scripts/collect-windows-logs.ps1:44]: I see. I mixed them up. Please ignore it.

**Pull request merged**